File: backend/AdminService/AdminService/Models/Appointment.cs
using System;
using System.Collections.Generic;

namespace AdminService.Models;

public partial class Appointment
{
    public int AppointmentId { get; set; }

    public int PatientId { get; set; }

    public int HospitalId { get; set; }

    public int SlotId { get; set; }

    public int DoseNumber { get; set; }

    public DateOnly BookingDate { get; set; }

    public TimeOnly BookingTime { get; set; }

    public string? Status { get; set; }

    public string? Remarks { get; set; }

    public virtual Hospital Hospital { get; set; } = null!;

    public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();

    public virtual Patient Patient { get; set; } = null!;

    public virtual Slot Slot { get; set; } = null!;
}

File: backend/patient-service/src/main/java/com/bharatteeka/patient/entity/Appointment.java
package com.bharatteeka.patient.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "appointment")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "appointment_id")
	private Integer appointmentId;

	@Column(name = "patient_id", nullable = false)
	private Integer patientId;

	@Column(name = "hospital_id", nullable = false)
	private Integer hospitalId;

	@Column(name = "slot_id", nullable = false)
	private Integer slotId;

	@Column(name = "dose_number", nullable = false)
	private Integer doseNumber;

	@Column(name = "booking_date", nullable = false)
	private LocalDate bookingDate;

	@Column(name = "booking_time", nullable = false)
	private LocalTime bookingTime;

	@Column(name = "status", length = 255)
	private String status;

	@Lob
	@Column(name = "remarks")
	private String remarks;
}
